actionpack/lib/action_controller/metal/rack_delegation.rb

```diff
@@ -8,9 +8,15 @@ module RackDelegation
     delegate :headers, :status=, :location=, :content_type=,
              :status, :location, :content_type, :response_code, :to => "@_response"
 
-    def dispatch(action, request)
+    module ClassMethods
```

Comment:
Not sure if this is the proper place to put this method. Also the name is not good enough. It is returning a new instance of a controller right?